protocol/0080-TBPM-time_based_pruning_mempool.md

```diff
@@ -0,0 +1,25 @@
+# Time based pruning of the mempool
+
+Transactions currently can stay in the mempool indefinitely. For non-validator commands, we want to be able to limit the time they are considered valid, once a certain amount of time has passed, the transaction should be rejected and removed from the mempool.
+
+## Network parameters
+
+- `network.transactions.maxttl` is a network parameter with type unsigned integer with the minimum value of `10` and a maximum value of `50`. This value determines the max number of blocks a given transaction is valid for. A transaction that specifies the `TTL`, this value *must* be less than, or equal to this parameter. If no `TTL` is specified, the transaction `TTL` will be set to this value.
+
+Part of the transaction data is the block height at which the transaction was created. Transactions will remain current until this block height + the `TTL`.
+
+## Including transactions
+
+This `TTL` is applied to all transactions, except for validator commands. Users have the ability to set the `TTL` manually, using transaction version 4, previous versions use the network parameter mentioned above as the `TTL`.
```

Comment:
Do we even need a new transaction version? If we default to the value of `network.transactions.maxttl` if its not set in the transaction then we're fine.